src/com/sap/piper/Notify.groovy

```diff
@@ -0,0 +1,51 @@
+package com.sap.piper
+
+import groovy.text.SimpleTemplateEngine
+
+class Notify implements Serializable {
+    private static enum Severity { ERROR, WARNING }
+    private final static String LIBRARY_NAME = 'piper-lib-os'
+    private final static String MESSAGE_PATTERN = '[${severity}] ${message} (${libName}/${stepName})'
+
+    protected static Utils instance = null
+
+    protected static Utils getUtilsInstance(){
+        instance = instance ?: new Utils()
+        return instance
+    }
+
+    static void warning(Map config, Script step, String msg, String stepName = null){
+        log(config, step, msg, stepName)
+    }
+
+    static void error(Map config, Script step, String msg, String stepName = null) {
+        log(config, step, msg, stepName, Severity.ERROR)
+    }
+
+    private static void log(Map config, Script step, String msg, String stepName, Severity severity = Severity.WARNING){
+        stepName = stepName ?: step.STEP_NAME
+        getUtilsInstance().pushToSWA([
```

Comment:
What is the reason for notifying about each time we log something? I think with this approach we get a lot of notifications. Is there any benefit associated with that?

Reply:
We will use the Notify when we need to get somebody informed. Thus it will help users to follow-up on actions and us to see how many leftovers are there ...
Important part when performing e.g. incompatible changes over time.

Reply:
I understand why we send notifications and I do not fight against this. But why should we send notification _always_ when logging something? That will end up in a huge amount of records we receive without corresponding benefit IMO.

Reply:
I can make the web analytics reporting optional. Would this be satisfying?

Reply:
> [optional]

Well, I mean I do not vote with "require changes", this I do only in cases where I think it will not work.

In general having bells and whistles in the code (we do what a method is supposed to do, **and** beside that we do optionally also x, y, z) leads to code which is harder to maintain in the future. Less is more. So I personally would feel more comfortable with dropping the notification as a side effect here in total.